src/core/IO/Handle.pm6

```diff
@@ -177,8 +177,17 @@ my class IO::Handle {
             $!decoder.set-line-separators(($!nl-in = $nl-in).list);
             $!encoder := $encoding.encoder(:translate-nl);
             $!encoding = $encoding.name;
+            # Add a byte order mark to the start of the file for utf16
+            nqp::if(nqp::iseq_s($!encoding, 'utf16'), (
+                if $create && !$exclusive && (!$append || $append && $!path.s == 0) {
+                  state $bom = buf16.new(0xFEFF);
```

Comment:
Is it better if this were a constant instead? `my constant $bom = buf16.new(0xFEFF);` ?

Reply:
I was mostly trying to make sure if you opened many files as utf16 it wouldn't have to create a new buf16 each time. If `my constant` will do that then that would be acceptable for me.

Reply:
> If my constant will do that then that would be acceptable for me.

Yeah, it'd create one at compile time and then re-use it for all the writes.

Reply:
Won't this get the `utf16` role get punned at compile time, and then create all sorts of extra startup overhead?

Reply:
@lizmat it only uses buf16 and not the utf16 role, though do you mean that would cause it to have to load anyway? I am thinking about using `blob16` instead of `buf16` since this is a constant and not designed to be modified.

Reply:
if it works, then it's ok with me.  I think we've abandoned reducing startup time at this point anyway.  I would use `buf16` as that is more likely to be used by users, so it would have already been punned for them.  See also src/core/core_epilogue about `buf8`

Reply:
@lizmat Deserialization works lazily, so in this case we won't trigger it unless the constant is used.